ddr/lib/ddr-blobgen/java/genBinaryBlob.cpp

```diff
@@ -227,10 +227,10 @@ JavaBlobGenerator::stringTableOffset(BlobHeader *blobHeader, J9HashTable *string
 	DDR_RC rc = DDR_RC_OK;
 	StringTableEntry exemplar;
 
-	*offset = UINT32_MAX;
+	*offset = U_32_MAX;
```

Comment:
While this works and is an ok change I think we should think about building non-compiler sources on z/OS with `-D_ISOC99_SOURCE` which I believe overcomes these issues. However such a change is outside the scope of this PR.

Reply:
In other parts of the code, we've been replacing uses of `UINT32_MAX` with `std::numeric_limits<uin32_t>::max()` from the `<limits>` header. It's pre-C++11 and I'm fairly certain it's supported by all our build compilers.

Reply:
Unfortunately `std::numeric_limits<uint32_t>::max()` does not work on z/OS unless `-D_ISOC99_SOURCE` which is currently not the case for non-compiler C++ sources which is why we need to look into this (outside the scope of this PR).

Reply:
Example:

```
[ 42%] Building CXX object ddr/lib/ddr-blobgen/CMakeFiles/omr_ddr_blobgen.dir/java/genBinaryBlob.cpp.o
"/trjit/sandbox/jeremif/developement/jitbuilder-zos/omr/ddr/lib/ddr-blobgen/java/genBinaryBlob.cpp", line 230.19: CCN5130 (S) "std::numeric_limits" is not declared.
```

Reply:
I see. :disappointed:

Reply:
I would prefer to add that option to the DDR code instead of changing the code

Reply:
I don't think it's as easy as adding the flag as we currently have no separation of "DDR only defines" so we would have to add it to the global defines, but then that causes problems elsewhere with with C++ vs C99.

That is really the direction we want to end up in, i.e. we use the same defines and same flags to build all of OMR. Unfortunately we are not there yet. I believe such a change is outside the scope of what we're trying to accomplish in this PR.

As a compromise I've created issue #3710 which will track this work.

Reply:
This got fixed by @keithc-ca in #3790.